### PR TITLE
[DOCS] - Minor updates to OStream Metrics exporter documentation

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #

## Changes

Minor updates to the OStream exporter example documentation
 - Add docs for Observable Instrument creation.
 - Remove reference to old metrics design docs.
 - Mention View creation as optional step, and view with default configuration is created by SDK if missing.
- small indentation changes.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/metrics_simple/README.md [hunk 1]
```diff
@@ -1,86 +1,108 @@
 # Simple Metrics Example
 
-This example initializes the metrics pipeline with 2 different instrument types.
-Here are more detailed explanations of each part.
+This example initializes the metrics pipeline with 3 different instrument types -
```
Comment (marcalff) [inline examples/metrics_simple/README.md:3]: '-' character at the end, did you mean ":" instead ?

File: examples/metrics_simple/README.md [hunk 2]
```diff
@@ -1,86 +1,108 @@
...
-The reader periodically collects metrics from the collector and exports them.
+The reader periodically collects metrics from the Aggregation Store and exports them.
 
-```cpp
-std::unique_ptr<metric_sdk::MetricExporter> exporter{new exportermetrics::OStreamMetricExporter};
-std::unique_ptr<metric_sdk::MetricReader> reader{
-    new metric_sdk::PeriodicExportingMetricReader(std::move(exporter), options)};
-```
+    ```cpp
+    std::unique_ptr<metric_sdk::MetricExporter> exporter{new exportermetrics::OStreamMetricExporter};
```
Comment (marcalff) [inline examples/metrics_simple/README.md:18]: Please add a
`  using namespace exportermetrics = xxx;`
so the reader knows exactly how to use this code.

File: examples/metrics_simple/README.md [hunk 3]
```diff
@@ -1,86 +1,108 @@
...
+    ```
+
+3. Optional: Create a view to map the Counter Instrument to Sum Aggregation.
+Add this view to provider. View creation is optional unless we want to add
+custom aggregation config, and attribute processor. Metrics SDK  will implicitly
+create a missing view with default mapping between Instrument and Aggregation.
+
+    ```cpp
+    std::unique_ptr<metric_sdk::InstrumentSelector> instrument_selector{
+        new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kCounter, "name_counter")};
```
Comment (marcalff) [inline examples/metrics_simple/README.md:39]: Likewise, using namespace metric_sdk = ???

Put all the using namespace statements at the beginning, for context.

File: examples/metrics_simple/README.md [hunk 4]
```diff
@@ -1,86 +1,108 @@
...
-
-auto histogram_counter = meter->CreateDoubleHistogram("histogram_name");
-auto context = opentelemetry::context::Context{};
-histogram_counter->Record(val, labelkv, context);
-```
-
-See [CONTRIBUTING.md](../../CONTRIBUTING.md) for instructions on building and
+    ```cpp
+    auto provider = std::shared_ptr<metrics_api::MeterProvider>(new opentelemetry::metrics::MeterProvider());
+    auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
```
Comment (marcalff) [inline examples/metrics_simple/README.md:28]: `new opentelemetry::metrics::MeterProvider()`

Does this builds ?
The API class is abstract.

How about instead:

    auto provider = std::shared_ptr<metrics_sdk::MeterProvider>(new opentelemetry::sdk::metrics::MeterProvider());

which also avoids the downcast from API to SDK.

**Review by marcalff — COMMENTED** [review]
Thanks for adding metrics documentation.

See a few questions / comments.

File: examples/metrics_simple/README.md (hunk 1 above)
Comment (lalitb) [reply examples/metrics_simple/README.md]: Yeah should have been ":", changed it.

File: examples/metrics_simple/README.md (hunk 2 above)
Comment (lalitb) [reply examples/metrics_simple/README.md:27]: Thanks. Added all the used aliases at the beginning.

File: examples/metrics_simple/README.md (hunk 3 above)
Comment (lalitb) [reply examples/metrics_simple/README.md]: done.

File: examples/metrics_simple/README.md (hunk 4 above)
Comment (lalitb) [reply examples/metrics_simple/README.md:37]: Good catch. Have changed it accordingly.

**Review by marcalff — APPROVED** [review]

**Pull request merged**